src/auth/middleware/basic/index.js

```diff
@@ -0,0 +1,28 @@
+'use strict';
+
+const bcrypt = require('bcrypt');
+const base64 = require('base-64');
+const { Users } = require('../../models')
+
+
+module.export = async (req, res, next) => {
+
+        let basicHeaderParts = req.headers.authorization.split(' ');
+        let encodedString = basicHeaderParts.pop();
+        let decodedString = base64.decode(encodedString);
+        let [username, password] = decodedString.split(':');
+
+        try {
+          const user = await Users.findOne({ where: { username: username } });
+          const valid = await bcrypt.compare(password, user.password);
+          if (valid) {
+            req.user = user;
+            next();
+            res.status(200).json(user);
+          }
+          else {
+            throw new Error('Invalid User');
+          }
+        } catch (error) { next('Invalid Login. message: ', error.message); }
```

Comment:
This is probably a situation where we don't want to send the error message back to the client; We can log it in the console, but especially when it comes to sensitive data like login information, we don't want to give any information other than "Invalid Login".